enhancements/helm3/namespace-scoped-helm-repository-crd.md

```diff
@@ -0,0 +1,201 @@
+  displayName: Public Helm stable charts
+  description: Public Helm stable charts hosted on HelmHub
+EOF
+
+$ kubectl get projecthelmchartrepositories --namespace my-namespace
+NAME          AGE
+stable        1m
+```
+
+The addition of namespace-scoped Helm repository does not impact behavior of the existing cluster-scoped Helm repository:
+
+```shell
+$ cat <<EOF | oc apply -f -
+apiVersion: helm.openshift.io/v1beta1
+kind: HelmChartRepository
+metadata:
+  name: stable
+spec:
+  url: https://kubernetes-charts.storage.googleapis.com
+  displayName: Public Helm stable charts
+  description: Public Helm stable charts hosted on HelmHub
+EOF
+
+$ cat <<EOF | oc apply --namespace my-namespace -f -
+apiVersion: helm.openshift.io/v1beta1
+kind: ProjectHelmChartRepository
+metadata:
+  name: incubator
+spec:
+  url: https://kubernetes-charts-incubator.storage.googleapis.com
+  displayName: Public Helm charts in incubator state
+EOF
+
+$ kubectl get helmchartrepositories
+NAME          AGE
+cluster       3h30m
+stable        1m
+
+$ kubectl get projecthelmchartrepositories --namespace my-namespace
+NAME          AGE
+incubator     1m
+```
+
+#### How would the UI render the namespace-scoped Helm repository
+
+As defined in [previous helm enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/helm3/console.md#how-would-the-ui-discover-the-charts), the UI will invoke the `/api/helm/charts/index.yaml` endpoint and the endpoint will proxy requests to the configured chart repositories.
+Enabling namespace-scoped Helm repository requires a new URL parameter `namespace` that specifies the namespace that contains the requested CRD.
+
+Then the chart repository proxy will use all configured chart repositories, including cluster-scoped and available namespace-scoped Helm repositories in the requested namespace, and deliver to the UI an aggregated index file. The UI then renders the developer catalog with the aggregated index file.
+
+### Implementation Details
+
+[ODC-5713](https://issues.redhat.com/browse/ODC-5713): In the OpenShift Dev Console, add capabilities that allow developers to create Helm chart registries which are namespace scoped, to enable developers to pull in their own Helm charts to their namespace.
+
+![Namespaced Helm Repo Console UI](../helm3/assets/namespace_helm_repo_console_ui.png)
+
+[HELM-244](https://issues.redhat.com/browse/HELM-244): Extend the `/api/helm/charts/index.yaml` API endpoint in OpenShift Dev Console backend to support query parameter `namespace`. For example, a GET request to `/api/helm/charts/index.yaml?namespace=foo` will respond an aggregated `index.yaml` file with entities extracted from both cluster scoped Helm repository and Helm repositories in `foo` namespace.
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+N/A
+
+### Test Plan
+
+N/A
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
```

Comment:
Is the API going to be GA right away? ( I'm fine with that - since we've had the Cluster-scoped API around for a while now ).

Either way, this section or the one above it deserves a mention.

Reply:
Agreed, added a sentence stating it's fully backward compatible, therefore will be released immediately.